game/world/objects/npc.cpp

```diff
@@ -169,7 +169,9 @@ Npc::Npc(World &owner, size_t instance, std::string_view waypoint)
   owner.script().initializeInstanceNpc(hnpc, instance);
   hnpc->wp       = std::string(waypoint);
 
-  // vanilla forces non-zero damage type
+  // vanilla behavior: equip best weapon and set non-zero damage type
+  if(!isMonster() && !isPlayer())
```

Comment:
One more question: `isMonster` is truly relevant? For wolfs/snappers should not mater; but maybe relevant for skeletons

Reply:
Yes best weapon autoequip works for skeletons and from what I tested it seems to only work for skeletons. Other monsters just have the weapon in their inventory.

For like a wolf weapon inserted with `CreateInvItem` is equipped and visible in OpenGothic, that's why I excluded monsters in general.

Reply:
After some thinking it's maybe best to prevent weapon equipping for monsters somewhere in the equip function directly in case someone adds a `EquipItem` call in script. 

Skeleton is detected based on guild in vanilla. Changing the guild of a golem to skeleton resulted in a draw/remove weapon stall.

Reply:
Since it not directly related to `isMonster` would be nice to at least understand how it intended to work.
I've done some testing with marvin - guild by itself seem to be irrelevant. Maybe it's related to `npc` having some builtin bones, such as `ZS_SWORD`/`ZS_RIGHTHAND` ?

Can you test what will happen if you assign `GIL_SHEEP` to skeleton or "Ske_Body" to sheep, while having weapon in inventory?

Reply:
UPD:
Tested by editing Nikolas guild: as `GIL_SKELETON` or `GIL_SHEEP` he still equips best weapon.

Reply:
> GIL_SHEEP to skeleton

Nothing.

> "Ske_Body" to sheep

Nothing but if `Mdl_SetVisual (self,	"HumanS.mds")`  with Sheep_Body sword is visible. 

For default sheep both `Npc_HasEquippedWeapon` and `Npc_HasEquippedMeleeWeapon` return true.

Reply:
> GIL_SHEEP to skeleton
Nothing.

You mean nothing changed, or skeleton does not equip weapons any more?)

Reply:
Nothing changed.